docs/deployment-ux.md

```diff
@@ -0,0 +1,410 @@
+    that don't care if another instance of the daemon is running on the
+    same host.
+
+    After deployment starts:
+
+    ``` {.shell}
+    $ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+    REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+    SomeEnvironment/Prod:1    Undeploying  0         5         5            5        
+    SomeEnvironment/Prod:2    Deploying    5         0         0            0        
+    ```
+
+    First task in new revision is launched, old revision is still
+    running everywhere:
+
+    ``` {.shell}
+    $ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+    REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+    SomeEnvironment/Prod:1    Undeploying  0         5         5            5        
+    SomeEnvironment/Prod:2    Deploying    5         1         1            1        
+    ```
+
+    First task in old revision is terminated:
+
+    ``` {.shell}
+    $ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+    REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+    SomeEnvironment/Prod:1    Undeploying  0         4         4            4        
+    SomeEnvironment/Prod:2    Deploying    5         1         1            1        
+    ```
+
+    Deployment progresses in a similar fashion until the last task of
+    the old revision is left:
+
+    ``` {.shell}
+    $ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+    REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+    SomeEnvironment/Prod:1    Undeploying  0         1         1            1        
+    SomeEnvironment/Prod:2    Deploying    5         5         5            5        
+    ```
+
+    Deployment completes once the final task in the old revision is
+    terminated:
+
+    ``` {.shell}
+    $ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+    REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+    SomeEnvironment/Prod:1    Inactive     0         0         0            0        
+    SomeEnvironment/Prod:2    Active       5         5         5            5        
+    ```
+
+Rollback to an earlier revision
+-------------------------------
+
+Let's say that revision 2 is actually broken, and we need to roll back
+to revision 1:
+
+``` {.shell}
+$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+SomeEnvironment/Prod:1    Inactive     0         0         0            0        
+SomeEnvironment/Prod:2    Active       5         5         5            5        
+```
+
+**Option 1**: Allow specifying earlier revisions to deploy. That
```

Comment:
We've decided that we'll go for Option 1, as it provides better insight into which environment revisions are actually running. With Option 2 (create a new environment revision to effect the rollback), we would no longer be able to tell that revision 1 and revision 3 are actually the same.

The tradeoff we'd be making, is that it's harder to see exactly when rollbacks happened. For example, if we repeatedly roll back to the same revision from multiple attempted deployments, all we'd see would be:

```
$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
SomeEnvironment/Prod:1    Active       5         5         5            5        
SomeEnvironment/Prod:2    Reverted     0         0         0            0    
SomeEnvironment/Prod:3    Reverted     0         0         0            0    
SomeEnvironment/Prod:4    Reverted     0         0         0            0    
```
In this case, we probably want to keep some sort of audit log around.